### ISSUE TITLE
Build Rust Toolchain on GitHub Windows 2022 runner - using shorter paths with subst

### ISSUE THREAD

**Issue by georgik** [esp-rs/rust-build#35]
Problem with long paths on Windows GitHub runner could be solved by following workaround:
```
 - name: Map local long path to new drive
        id: map_path
        shell: pwsh
        run: subst "p:" "$env:GITHUB_WORKSPACE"
```

**Comment by georgik** [comment]
Problem with command: `build --stage 2` results on building stage 3
https://rustc-dev-guide.rust-lang.org/building/bootstrapping.html#understanding-stages-of-bootstrap

[Building vs. running](https://rustc-dev-guide.rust-lang.org/building/bootstrapping.html#building-vs-running)
Note that build --stage N compiler/rustc does not build the stage N compiler: instead it builds the stage N+1 compiler using the stage N compiler.

**Comment by georgik** [comment]
Solved by #41
